pkgs/by-name/li/libhyper/package.nix

```diff
@@ -0,0 +1,92 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  rustPlatform,
+  clientSupport ? full,
+  full ? true,
+  http1Support ? full,
+  http2Support ? full,
+  serverSupport ? full,
+  tracingSupport ? false,
+}:
+
+let
+  isFull = full || clientSupport && http1Support && http2Support && serverSupport;
+in
+rustPlatform.buildRustPackage rec {
+  pname = "libhyper";
+  version = "1.5.1";
+
+  src = fetchFromGitHub {
+    owner = "hyperium";
+    repo = "hyper";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-neFCZfMWm+FV+VlSXpvrxaGj+wF+exVDZv3JG4wiX7o=";
+  };
+
+  # Add a Cargo.lock
+  #
+  # Make sure rustc always emits cdylib artifacts
+  postPatch = ''
+    ln -sf ${./Cargo.lock} Cargo.lock
```

Comment:
Upstream does not commit their lockfiles?

I think a better approach would be to ask upstream to commit their lockfile, mentioning our reproducibility requirements which needs the lockfile to compare the hash with.

Reply:
Hyper is usually used as a library, and Cargo's commit guidelines generally advise against including a Cargo.lock for libraries

Reply:
Opened https://github.com/hyperium/hyper/issues/3788